src/crons/cache.warmer/cache.warmer.service.ts

```diff
@@ -335,4 +366,30 @@ export class CacheWarmerService {
   private async refreshCacheKey(key: string, ttl: number) {
     await this.clientProxy.emit('refreshCacheKey', { key, ttl });
   }
+
+  private async getTransactionCountForInterval(address: string, startDate: Date, endDate: Date): Promise<number> {
+    const after = startDate.getTime();
+    const before = endDate.getTime();
+
+    console.log(`Interval: ${new Date(after).toISOString()} - ${new Date(before).toISOString()}`);
+
+    const filter: TransactionFilter = { before, after };
+
+    console.log(`Before value: ${filter.before} and After value: ${filter.after}`);
+    const result = await this.indexerService.getTransactionCount(filter, address);
+
+    return result;
+  }
+
+  private subtractDays(date: Date, days: number): Date {
```

Comment:
we could have a single function, something like:
```
private subtractTimeUnit(date: Date, value: number, unit: 'hours' | 'days'): Date {
  const result = new Date(date);
  
  if (unit === 'hours') {
    result.setHours(result.getHours() - value);
  } else if (unit === 'days') {
    result.setDate(result.getDate() - value);
  } else {
    throw new Error('Invalid time unit specified');
  }
  
  return result;
}
```

or we can use `Constants.OneDay/OneHour` and express everything in seconds.